File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        idx_map={}
        arr= [1]*len(nums)
        for x in range(len(nums)):
            idx_map[nums[x]]=x
        fans = []
        def ps(ans,arr):
            f=True
            for x in nums:
                if arr[idx_map[x]]:
                    f=False
                    break
            if f:
                fans.append(ans.copy())
                return


            for x in nums:
                if arr[idx_map[x]]:
                    arr[idx_map[x]]=0
                    ans.append(x)
                    ps(ans,arr)
                    arr[idx_map[x]]=1
                    ans.pop()
        ps([],arr)
        return fans

File: Data Structures & Algorithms/concatenation-of-array/submission-0.cs
public class Solution {
    public int[] GetConcatenation(int[] nums) {
        int[] arr=new int[2*nums.Length];
        // Console.WriteLine("value");
        for(int i=0;i<nums.Length;i++){
            arr[i]=nums[i];
            arr[i+nums.Length]=nums[i];
        }
        return arr;
    }
}